src/java.desktop/macosx/native/libawt_lwawt/awt/AWTWindow.m

```diff
@@ -288,6 +288,9 @@ - (id) initWithPlatformWindow:(jobject)platformWindow
 {
 AWT_ASSERT_APPKIT_THREAD;
 
+    if (@available(macOS 10.12, *)) {
+        [NSWindow setAllowsAutomaticWindowTabbing:NO];
```

Comment:
This works as long as you are OK with requiring an Xcode with the 10.12 SDK to build (it won't affect the minimum runtime version). I think this is fine, but wanted to point that out.